core/modules/qdisp/ResponseRequester.h

```diff
@@ -101,8 +109,10 @@ class ResponseRequester {
         boost::shared_ptr<CancelFunc> f;
         {
             boost::lock_guard<boost::mutex> lock(_cancelMutex);
-            f.swap(_cancelFunc);
-            _cancelled = true;
+            if(!_cancelled) {
+                f = _cancelFunc;
```

Comment:
The swap() call above nulls out _cancelFunc.  The new code doesn't do this.  Was that an intentional change?